versions/development/SPEC.md

```diff
@@ -760,7 +760,9 @@ Engines should at the very least support the following protocols for import URIs
 
 * `http://` and `https://`
 * `file://`
-* no protocol (which should be interpreted as `file://`
+* No protocol (see below)
+
+In the event that there is no protocol, if this is an absolute path (i.e. starts with `/`) it is interpreted as a `file://`. Otherwise it is **relative** to the location of the current document. For instance, if the current doc is at `file://foo/bar/baz/qux.wdl` and specificies `import some/task.wdl` this will be resolved to `file://foo/bar/baz/some/task.wdl`. Likewise if the current doc is at `http://www.github.com/openwdl/coolwdls/myWorkflow.wdl` and imports `subworkflow.wdl`, it will be resolved to `http://www.github.com/openwdl/coolwdls/subworkflow.wdl`. It is up to the implementation to provide a mechanism which will allow these imports to be resolved correctly.
```

Comment:
what about `import ../../quux/quuz.wdl`? allowed?

Reply:
IMO as long as that makes sense relative to the "current" URL, I don't see why not (although I would personally still like some scheme like https://github.com/openwdl/wdl/issues/214 to indicate how to get back to the "import root" without resorting to long trails of `../`)

Reply:
@yfarjoun So @cjllanwarne and I have talked about this a bit, and a similar discussion was had in Toronto more generally among Cloud Workstream members (so applying to things beyond WDL as well). The sentiment from the latter (shared by myself) is that people **shouldn't** use `../` but they're **going** to so don't explicitly disallow it. IOW make it one of those best practices/linter type of things.

Why shouldn't they? It assumes POSIX-like file structures, and that might fall apart at some point in some arena. However, for practical purposes it will work, so ....

Reply:
@geoffjentry I agree that we cannot explicitly restrict the use of `../` paths. as annoying as that is. People will undoubetly use it, but I like the approach you are taking. we will support but only grudgingly. We can clearly describe in best practices what way users SHOULD do this